### PR TITLE
Move BMC options and help text to bmc_util [depends: 3565]

### PR DESCRIPTION
**Pull request by peterschrammel**
Based on #3565, review only last commit

- [X] Each commit message has a non-empty body, explaining why the change was made.
- [ ] Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- [ ] The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- [ ] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- [ ] My commit message includes data points confirming performance improvements (if claimed).
- [X] My PR is restricted to a single feature or bugfix.
- [ ] White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

**Review by allredj — COMMENTED** [review]
:no_entry_sign:
This PR failed Diffblue compatibility checks (cbmc commit: bb31b47d494b53b79010521bb60d3bd4ba3c1c54).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/94899676
Status will be re-evaluated on next push.
Please contact @peterschrammel, @thk123, or @allredj for support.

Common spurious failures:
  - the cbmc commit has disappeared in the mean time (e.g. in a force-push)
  - the author is not in the list of contributors (e.g. first-time contributors).

The incompatibility may have been introduced by an earlier PR. In that case merging this
PR should be avoided unless it fixes the current incompatibility.

File: jbmc/src/jbmc/jbmc_parse_options.cpp
Comment (tautschnig) [inline jbmc/src/jbmc/jbmc_parse_options.cpp:56]: Would you mind adding an issue or are you tracking this in some way?

**Review by tautschnig — APPROVED** [review]

File: jbmc/src/jbmc/jbmc_parse_options.cpp
Comment (peterschrammel) [reply jbmc/src/jbmc/jbmc_parse_options.cpp:56]: That will be the result as soon as I reach the top of this DAG of PRs.

**Review by smowton — APPROVED** [review]

**Comment by tautschnig** [comment]
Marking do-not-merge until depended-on PR is merged.

**Comment by peterschrammel** [comment]
> Marking do-not-merge until depended-on PR is merged.

Can we assume that "dependent" implies "do not merge" to reduce the number of labels?
Or rename "dependent" to "dependent - do not merge".

**Comment by tautschnig** [comment]
> Or rename "dependent" to "dependent - do not merge".

+1 on this - I'd like to avoid implicit information.

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: 8d150f01b466adba6278f6d7412a5b5ac2aff52f).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/96787768

**Pull request merged**
